fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -3272,4 +3272,68 @@ public IndexState needRebuildIndex(Index index, long recordCount, boolean indexO
             assertTrue(recordStore.getRecordStoreState().getStoreHeader().getOmitUnsplitRecordSuffix());
         }
     }
+
+    @Test
+    public void testIndexStateFormatVersionUpgrade() {
+        final RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestNoIndexesProto.getDescriptor());
+        try (FDBRecordContext context = openContext()) {
+            recordStore = FDBRecordStore.newBuilder()
+                    .setContext(context)
+                    .setMetaDataProvider(builder)
+                    .setKeySpacePath(path)
+                    .setFormatVersion(FDBRecordStore.SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION - 1)
+                    .create();
+            assertEquals(FDBRecordStore.SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION - 1, recordStore.getFormatVersion());
+            // Save 200 records without any indexes.
+            for (int i = 0; i < 200; i++) {
+                TestNoIndexesProto.MySimpleRecord record = TestNoIndexesProto.MySimpleRecord.newBuilder()
+                        .setRecNo(i).setNumValue(i + 1000).build();
+                recordStore.saveRecord(record);
```

Comment:
Instead of adding 200 records, this could instead call `markIndexWriteOnly`.

My hesitation here is that writing records takes time, and it would be nice if tests could be fast.

Reply:
sure. done.